src/index.tsx

```diff
@@ -140,6 +140,7 @@ export class URLSearchParams {
 
 export class URL {
   private _url: ReturnType<FastUrlNative['URL']>;
+  private _searchParams: URLSearchParams;
```

Comment:
A lot of the changes to this code are so that the integration tests with URLSearchParams pass, since updating url.searchParams should update `url.href`, `url.search`, `url.toString()`, and `url.toJSON()`. This implementation is roughly based off of node's own implementation of URL, found [here](https://github.com/nodejs/node/blob/64c6d97463c29bade4d6081683dab2cd7cda298d/lib/internal/url.js#L772)

Reply:
See these tests: https://github.com/KusStar/react-native-fast-url/pull/2/files#diff-54ebcb444ef4a96fb1ac9fc89de583e761d4e0060fc1fa0c658bcb61bdd6da09R159